src/utils/env/index.ts

```diff
@@ -102,15 +102,18 @@ export const getValueForContext = (
    */
   contextOrBranch: ContextOrBranch,
 ): EnvelopeEnvVarValue | undefined => {
-  const valueForContext = values.find((val) => {
-    const isSupportedContext = (SUPPORTED_CONTEXTS as readonly string[]).includes(contextOrBranch)
-    if (!isSupportedContext) {
-      // the "context" option passed in is actually the name of a branch
-      return val.context === 'all' || val.context_parameter === contextOrBranch
+  const isSupportedContext = (SUPPORTED_CONTEXTS as readonly string[]).includes(contextOrBranch)
+  if (!isSupportedContext) {
+    const valueMatchingAsBranch = values.find((val) => val.context_parameter === contextOrBranch)
+    // This is a `branch` context, which is an override, so it takes precedence
+    if (valueMatchingAsBranch != null) {
+      return valueMatchingAsBranch
     }
-    return val.context === 'all' || val.context === contextOrBranch
-  })
-  return valueForContext ?? undefined
+    const valueMatchingContext = values.find((val) => val.context === 'all' || val.context === 'branch-deploy')
```

Comment:
I'm fairly certain there's another bug here (which I haven't changed the behaviour of at all in this PR), where whether the `all` context value matches or the one that should be overriding does depends on the order of the elements returned from the API. I assume it's returned in order of precedence and so just happens to not exhibit the bug? In any case, I'll open a follow-up PR later.